packages/react-relay/modern/ReactRelayQueryRenderer.js

```diff
@@ -96,10 +97,26 @@ class ReactRelayQueryRenderer extends React.Component {
     };
     this._rootSubscription = null;
     this._selectionReference = null;
-    if (query) {
-      this.state = {
-        readyState: getDefaultState(),
-      };
+
+    if (operation) {
+      if (lookup && environment.check(operation.root)) {
+        // data is available in the store, render without making any requests
+        const snapshot = environment.lookup(operation.fragment);
+        this.state = {
+          readyState: {
+            error: null,
+            props: snapshot.data,
+            retry: () => {
+              this._fetch(operation, props.cacheConfig);
```

Comment:
I'm not sure why flow is failing here. I think it thinks that operation could potentially be `null`, but it is guarded by the `if (operation) {` above.

```
packages/react-relay/modern/ReactRelayQueryRenderer.js:110
110:               this._fetch(operation, props.cacheConfig);
                               ^^^^^^^^^ null. This type is incompatible with the expected param type of
220:   _fetch(operation: OperationSelector, cacheConfig: ?CacheConfig): void {
                         ^^^^^^^^^^^^^^^^^ object type
```